Source/ProgressionSystemRuntime/Private/Widgets/PSEndGameWidget.cpp

```diff
@@ -35,38 +37,40 @@ void UPSMenuWidget::NativeConstruct()
 }
 
 // Called when the end game state was changed to toggle progression widget visibility
-void UPSMenuWidget::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+void UPSEndGameWidget::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
 {
 	switch (CurrentGameState)
 	{
 	case ECurrentGameState::GameStarting:
 		SetVisibility(ESlateVisibility::Collapsed);
 		break;
+	case ECurrentGameState::Menu:
```

Comment:
since both `case` do the same, please do `fallthrough`:
```cpp
        case ECurrentGameState::Menu: // Fallthrough
	case ECurrentGameState::GameStarting:
		SetVisibility(ESlateVisibility::Collapsed);
		break;
        default: break;
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/43/commits/eab433dd1885b9fd32e1d2e33091ae42b6b18ad1

Reply:
@h4rdmol, is not finished yet, compare my example and you logic. Also it's good practive to write `// Fallthrough` comment showing others you did it intentionally

Reply:
added https://github.com/h4rdmol/ProgressionSystem/pull/43/commits/c6fb2fa1c22580ec9029ef3ae3b19fa6e2461476